### PR TITLE
LibGfx+icc: Implement half of lutAToBType and lutBToAType

### PR DESCRIPTION
**Pull request by nico**
These are among the permitted tag types of ATo0Tag and BToA0Tag, which are among the required tags of most profiles. They are the last permitted tag types for those profiles (the other are lut8Type or lut16Type, which are already implemented).

They are pretty chonky types though, so this only implements support for the E matrix and the CLUT. Support for the various curves will be in a future PR.

### CODE REVIEW

**Comment by nico** [comment]
Screenshot for https://littlecms.com/img/blog/stress.jpg :

```
...
AToB0Tag ('A2B0'): type 'mAB ', offset 400, size 1968
    3 input channels, 3 output channels
    color lookup table: 11 x 5 x 5 grid points, 825 u16 entries
    e = [ 1, 0, 0, 0,
          0, 1, 0, 0,
          0, 0, 1, 0 ]
BToA0Tag ('B2A0'): type 'mBA ', offset 2368, size 1968
    3 input channels, 3 output channels
    e = [ 1, 0, 0, 0,
          0, 1, 0, 0,
          0, 0, 1, 0 ]
    color lookup table: 11 x 5 x 5 grid points, 825 u16 entries
```

File: Userland/Libraries/LibGfx/ICC/TagTypes.cpp [hunk 1]
```diff
@@ -181,6 +204,209 @@ ErrorOr<NonnullRefPtr<Lut8TagData>> Lut8TagData::from_bytes(ReadonlyBytes bytes,
...
+
+    if (header.number_of_input_channels >= sizeof(CLUTHeader::number_of_grid_points_in_dimension))
+        return Error::from_string_literal("ICC::Profile: clut has too many input channels");
+
+    auto& clut_header = *bit_cast<CLUTHeader const*>(bytes.data() + header.offset_to_clut);
+
+    // "Number of grid points in each dimension. Only the first i entries are used, where i is the number of input channels."
+    Vector<u8, 4> number_of_grid_points_in_dimension;
+    TRY(number_of_grid_points_in_dimension.try_resize(header.number_of_input_channels));
+    for (unsigned i = 0; i < header.number_of_input_channels; ++i)
```
Comment (linusg) [inline Userland/Libraries/LibGfx/ICC/TagTypes.cpp:221]: using `unsigned` for loop counters seems unusual, what's wrong with `size_t`? Same below, and especially for comparisons with `sizeof` results.

File: Userland/Libraries/LibGfx/ICC/TagTypes.cpp [hunk 2]
```diff
@@ -181,6 +204,209 @@ ErrorOr<NonnullRefPtr<Lut8TagData>> Lut8TagData::from_bytes(ReadonlyBytes bytes,
...
+    // "There are the same number of “A” curves as there are output channels. The “A” curves may only be used when
+    //  the CLUT is used. The curves are stored sequentially, with 00h bytes used for padding between them if needed.
+    //  Each “A” curve is stored as an embedded curveType or a parametricCurveType (see 10.5 or 10.16). The length
+    //  is as indicated by the convention of the proper curve type. Note that the entire tag type, including the tag type
+    //  signature and reserved bytes, is included for each curve."
+    if (header.offset_to_a_curves) {
+        // FIXME
+    }
+
+    // FIXME
```
Comment (linusg) [inline Userland/Libraries/LibGfx/ICC/TagTypes.cpp:406]: Fix what?

**Comment by nico** [comment]
Thanks! All done, ptal.

**Comment by nico** [comment]
Derp, sorry. Done now, I think.

**Comment by nico** [comment]
(The whole sad story was that I didn't see your comments, deleted all my local branches, then noticed the comments on this one, found the old commit in `git reflog`, checked it out, checked out a new local branch, but then typod the branch name and the changes did end up in a branch on my fork, but not on the one associated with this PR. But fixed now.)

**Pull request merged**